components/canvas/WhiteboardClient.tsx

```
@@ -1,64 +1,725 @@
+        if (currentElementRef.current) {
+            drawElement(ctx, currentElementRef.current);
+        }
+
+        ctx.restore();
+    }, [drawElement]);
+
+    // ─── Animation loop ────────────────────────────────────────
+    const animFrameRef = useRef<number>(0);
+    const renderLoop = useCallback(() => {
+        render();
+        animFrameRef.current = requestAnimationFrame(renderLoop);
```

ESLint

[error] Error: Cannot access variable before it is declared

`renderLoop` is accessed before it is declared, which prevents the earlier access from updating when this value changes over time.

/home/jailuser/git/components/canvas/WhiteboardClient.tsx:277:54
  275 |     const renderLoop = useCallback(() => {
  276 |         render();
> 277 |         animFrameRef.current = requestAnimationFrame(renderLoop);
      |                                                      ^^^^^^^^^^ `renderLoop` accessed before it is declared
  278 |     }, [render]);
  279 |
  280 |     useEffect(() => {

/home/jailuser/git/components/canvas/WhiteboardClient.tsx:275:5
  273 |     // ─── Animation loop ────────────────────────────────────────
  274 |     const animFrameRef = useRef<number>(0);
> 275 |     const renderLoop = useCallback(() => {
      |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 276 |         render();
      | ^^^^^^^^^^^^^^^^^
> 277 |         animFrameRef.current = requestAnimationFrame(renderLoop);
      | ^^^^^^^^^^^^^^^^^
> 278 |     }, [render]);
      | ^^^^^^^^^^^^^^^^^^ `renderLoop` is declared here
  279 |
  280 |     useEffect(() => {
  281 |         animFrameRef.current = requestAnimationFrame(renderLoop);

(react-hooks/immutability)